apps/sparo-lib/src/cli/commands/pull.ts

```diff
@@ -69,7 +70,12 @@ export class PullCommand implements ICommand<IPullCommandOptions> {
     await this._gitRemoteFetchConfigService.pruneRemoteBranchesInGitConfigAsync(remote || 'origin');
 
     // invoke native git pull command
-    gitService.executeGitCommand({ args: pullArgs });
+    const pullProcess: SpawnSyncReturns<string> = gitService.executeGitCommand({ args: pullArgs });
+
+    if (pullProcess.status !== 0) {
+      // Pull failed
+      throw new Error('"git pull" operation failed');
```

Comment:
Suggest to include the process exit code in this message (`"git pull" operation failed (exit code #123)`), or else have Sparo return the same exit code.